### ISSUE TITLE
Unable to create archive

### ISSUE THREAD

**Issue by saivul** [Studio-42/elFinder#3042]
![Screenshot from 2019-11-28 10-48-40](https://user-images.githubusercontent.com/30980269/69775887-d7381980-11cc-11ea-91cb-99cc20425d78.png)

Elfinder 2.1.50
Codeigniter with CSRF true. upload and put working fine.

**Comment by nao-pon** [comment]
@saivul Requires a PHP Zip extension or zip command on the server side.

**Comment by saivul** [comment]
zip is already running well in my codeigniter application

**Comment by TechCatOther** [comment]
It can be solved by changing tmp directory in connector. Problem is hardlinks on zipdl function. This function create hardlink in php folder and absoulite path of file in files folder. If you have file on other devices and elFinder not on same device this error will be show.

Anyway it is not good solution to creating hardlink in php/.tmp folder. We need to move it to files directory to resolve issuse.

**Comment by nao-pon** [comment]
@saivul , @TechCatOther First, try the link function. If it fails, the file pointer is used to copy the contents. Certainly the error is reported only once, but the process should be fine.

There may be bugs that I don't know about. Let me know if you find anything.

**Comment by TechCatOther** [comment]
Yep, I fix a bug localy with symlinks archive this. When different device using for files folder and elfinder self link was broken and in archive symlink was copy of files, not symlink self.

I don't know. May be this is not a bug for windows users, but in *nix systems backuping was broken.

**Comment by nao-pon** [comment]
OK, I've made a fix to make the copying of the file more strict and close it. If you still have the same problem, reopen it or create a new issue.
